### ISSUE TITLE
Some questions regarding your paper

### ISSUE THREAD

**Issue by Jiahao000** [yossigandelsman/test_time_training_mae#2]
Thanks for sharing your great work! After reading your paper, I have several questions and hope you could help me clarify:

- The paper states that "we use MAE based on the ViT-Large encoder, pre-trained for 800 epochs on ImageNet-1k." However, the [script](https://github.com/yossigandelsman/test_time_training_mae#training-mae) provided here only trains for 90 epochs, with some other configs like `blr`, `warmup_epochs` `batch_size`, etc., different from the MAE [official repo](https://github.com/facebookresearch/mae/blob/main/PRETRAIN.md#pre-training-mae). May I know why they are different?
- During training-time training,  the paper says "following the protocol in MAE for linear probing", while the [script](https://github.com/yossigandelsman/test_time_training_mae#training-the-classification-head) provided use a different training schedule compared with the MAE [offical repo](https://github.com/facebookresearch/mae/blob/main/FINETUNE.md#linear-probing), e.g., `epochs`, `blr`, `weight_decay`, `batch_size`.  May I know the reasons to choose these different hyper-parameters?
- To my understanding, the use of pre-trained MAE model naturally provides a good initialization for the shared encoder and the SSL decoder for subsequent test-time training. What if other pre-trained models are used? In this case, the tasks during pre-training stage and test-time training stage could be different. If the ViT probing is used without joint training, will the performance be degraded significantly?

Thanks in advance!

**Comment by yossigandelsman** [comment]
Thank you for your questions! 

* You are absolutely right about it, we will change the script to the official repo hyperparameters.
* We found those parameters to work better in our setting, but the differences are not very big. The results in the paper are for the provided checkpoint, which was trained with the script hyperparmeters. 
* We didn't use joint-training here. We just ViT-probed a pretrained MAE. We did not try what you suggested, but we believe that using a different SSL task during test-time can create a large shift in the distribution of the features after the test-time training, which may cause the ViT-head to not generalize well. Yet, it is an interesting question to explore.
